### PR TITLE
Tests for index endpoint of devices api and adding reporter for jasmine

### PR DESCRIPTION
**Pull request by kapil-ms**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kapil-ms__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (kaagarw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: apitests/spec/devices_spec.ts [hunk 1]
```diff
@@ -30,5 +30,96 @@ describe('devices api', () => {
                 done();
             });
         });
+
+        it('should always have these properties', (done) => {
```
Comment (rraomsft) [inline apitests/spec/devices_spec.ts]: Rename: it should always have required properties

File: apitests/spec/devices_spec.ts [hunk 2]
```diff
@@ -30,5 +30,96 @@ describe('devices api', () => {
...
+                expect(result.data[0]._rid).toBeTruthy();
+                expect(result.data[0]._self).toBeTruthy();
+                expect(result.data[0]._etag).toBeTruthy();
+                expect(result.data[0]._ts).toBeTruthy();
+                expect(result.data[0]._attachments).toBeTruthy();
+                done();
+            });
+        })
+
+        it('should have these attributes for enabled device', (done) => {
```
Comment (rraomsft) [inline apitests/spec/devices_spec.ts]: Rename: it should have required attributes for enabled devices

File: apitests/spec/devices_spec.ts [hunk 3]
```diff
@@ -30,5 +30,96 @@ describe('devices api', () => {
...
+            request.get('', (err, resp, result) => {
+                expect(result.data[1].DeviceProperties.HubEnabledState).not.toBeNull();
```
Comment (rraomsft) [inline apitests/spec/devices_spec.ts]: This is going to be brittle. If the order changes, e.g. someone adds a new device, there's no guarantee result.data[1].DeviceProperties.HubEnabledState will not be null. You'll want to first do a find of any device in result.data that is hub enabled and then validate the properties below

File: apitests/spec/devices_spec.ts [hunk 4]
```diff
@@ -24,11 +54,108 @@ describe('devices api', () => {
...
+                expect(result.data[0]._etag).toBeTruthy();
+                expect(result.data[0]._ts).toBeTruthy();
+                expect(result.data[0]._attachments).toBeTruthy();
+                done();
+            });
+        })
+
+        it('should have required attributes for enabled devices', (done) => {
+            request.get('', (err, resp, result:Devices) => {
+                let device:DeviceInfo = findEnabledDevice(result);
```
Comment (rraomsft) [inline apitests/spec/devices_spec.ts]: expect device to be a valid object to guard against case where no enabled devices are found.

**Pull request merged**